src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -166,30 +168,14 @@ public SNITCPHandle(string serverName, int port, long timerExpire, object callba
 
         private static async Task<Socket> ConnectAsync(string serverName, int port)
         {
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-            {
-                var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
-                await socket.ConnectAsync(serverName, port).ConfigureAwait(false);
-                return socket;
-            }
-
-            // On unix we can't use the instance Socket methods that take multiple endpoints
-
-            IPAddress[] addresses = await Dns.GetHostAddressesAsync(serverName).ConfigureAwait(false);
-            return await ConnectAsync(addresses, port).ConfigureAwait(false);
+            Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            await socket.ConnectAsync(serverName, port).ConfigureAwait(false);
```

Comment:
This is going to fail on Unix if serverName is anything other than an IP address.